src/parse/MissionParse.cpp

```diff
@@ -84,7 +84,19 @@ bool MissionParse::parse(const std::string &filename) {
     std::regex reg("\\$\\{.+?=(.+?)\\}");
     content = std::regex_replace(content, reg, fmt);
 
-    doc.parse<0>(&content[0]);
+    //Parse the xml tree.
+    //doc.parse requires a null terminated string
+    std::vector<char> contentvec(content.size() + 1); //allocation done here
+    contentvec.assign(content.begin(), content.end()); //copy
+    contentvec.push_back('\0'); //shouldn't reallocate
+    try {
+      //Note: This parse function can hard fail (seg fault, no exception) on
+      //      badly formatted xml data. Sometimes it'll except, sometimes not.
+      doc.parse<0>(contentvec.data());
+    } catch (...) {
+      cout << "scrimmage::MissionParse::parse: Exception during rapidxml::xml_document<>.parse<>()." << endl;
```

Comment:
Can we throw an error here instead of using cout?

Reply:
It is already. It returns false, that's the error mechanism used here.